github.go

```diff
@@ -52,59 +55,69 @@ func runGitHubHandler(c *gin.Context) {
 		},
 	}
 
-	importer.Token, err = importer.getToken(i)
+	importer.Token, err = importer.getToken(c, i)
 	if err != nil {
-		c.Error(err)
-		c.JSON(http.StatusBadRequest, fmt.Sprintf("Unable to get token. %+v", err))
 		return
 	}
 
 	importer.Provider, err = importer.getProvider(i, importer.Token)
 	if err != nil {
-		c.Error(err)
-		c.JSON(http.StatusBadRequest, fmt.Sprintf("Unable to get provider. %+v", err))
+		ginutil.WriteAPIClientReadError(c, err,
+			fmt.Sprintf("Unable to get GitHub provider by ID %v or name %q", i.ProviderID, i.Provider))
 		return
 	}
 
 	importer.GithubClient, err = importer.initGithubConnection()
 	if err != nil {
-		c.Error(err)
-		c.JSON(http.StatusBadRequest, fmt.Sprintf("Unable to init github connection. %+v", err))
+		ginutil.WriteAPIClientReadError(c, err,
+			fmt.Sprintf("Unable to parse provider url %q or upload url %q",
+				importer.Provider.URL, importer.Provider.UploadURL))
 		return
 	}
 
 	if i.ProjectID != 0 || i.Project != "" {
 		err = importer.importProject(i)
 		if err != nil {
-			c.Error(err)
-			c.JSON(http.StatusBadRequest, fmt.Sprintf("Unable to import project. %+v", err))
+			ginutil.WriteAPIClientWriteError(c, err,
+				fmt.Sprintf("Unable to import project %q (id: %v) from GitHub.", i.Project, i.ProjectID))
 			return
 		}
 	} else {
 		err = importer.importGroup(i.Group)
 		if err != nil {
-			c.Error(err)
-			c.JSON(http.StatusBadRequest, fmt.Sprintf("Unable to import organization or group. %+v", err))
+			ginutil.WriteAPIClientWriteError(c, err,
+				fmt.Sprintf("Unable to import group %q from GitHub.", i.Group))
 			return
 		}
 	}
 
 	c.Status(http.StatusCreated)
 }
 
-func (importer githubImporter) getToken(i importBody) (wharfapi.Token, error) {
+func (importer githubImporter) getToken(c *gin.Context, i importBody) (wharfapi.Token, error) {
```

Comment:
Suggest renaming this method to state that it's writing problems such as `getTokenWritesProblem`, and suggest returning a boolean instead of an error as the errors are handled already.